libs/langgraph/langgraph/_api/deprecation.py

```diff
@@ -1,38 +1,64 @@
 import functools
 import warnings
-from typing import Any, Callable, TypeVar, cast
+from typing import Any, Callable, Type, TypeVar, Union, cast
 
 
 class LangGraphDeprecationWarning(DeprecationWarning):
     pass
 
 
 F = TypeVar("F", bound=Callable[..., Any])
+C = TypeVar("C", bound=Type[Any])
 
 
 def deprecated(
```

Comment:
you've confirmed this decorator doesn't strip types?

Reply:
yup